amalgamate

```diff
@@ -674,7 +674,8 @@ file-printf-prepend()
 	local format=$2
 	local arguments=("${@:3}")
 
-	printf '%s' "$(printf "$format" "${arguments[@]}"; printf '%s' $(cat "$file_path"))" > "$file_path"
+	content=$(cat "$file_path")
+	printf "$format%s" "${arguments[@]}" "$content" > "$file_path"
```

Comment:
I would suggest solution that should not load the whole file into memory (despite realistically this wouldn't cause any issues), though a bit more verbose – writing prepending string first into temporary file, and then redirecting cat into that temporary file.

```
local tmp_file_path="$file_path.tmp"
mktemp "$tmp_file_path" &> /dev/null

printf "$format" "${arguments[@]}" > "$tmp_file_path"
cat "$file_path" >> "$tmp_file_path"
rm "$file_path"
mv "$tmp_file_path" "$file_path"
```

Reply:
Perfect, thanks a lot! Could you just commit your fix, then I would close this PR. Thx a lot!